### PR TITLE
Update modules.lua

### PR DESCRIPTION
**Pull request by luanluciano93**
I have doubts in two functions:
• doAddContainerItem(container, itemid, subType) - Container(container):addItemEx(Item(itemid)) ?
• player:say(text, type, false, pcid, player:getPosition()) - pcid or player2 ?

### CODE REVIEW

File: data/npc/lib/npc.lua [hunk 1]
```diff
@@ -11,33 +11,34 @@ function msgcontains(message, keyword)
...
 	local amount = amount or 1
 	local subType = subType or 0
 	local item = 0
-	if isItemStackable(itemid) then
+	if ItemType(itemid):isStackable() then
 		if inBackpacks then
-			stuff = doCreateItemEx(backpack, 1)
-			item = doAddContainerItem(stuff, itemid, math.min(100, amount))
+			stuff = Game.createItem(backpack, 1)
+			item = doAddContainerItem(stuff, itemid, math.min(100, amount)) -- ???
```
Comment (WibbenZ) [inline data/npc/lib/npc.lua]: Why did you add "-- ???"
And why did you not change the compat function? https://github.com/otland/forgottenserver/blob/master/data/lib/compat/compat.lua#L583-L594

File: data/npc/lib/npc.lua [hunk 2]
```diff
@@ -11,33 +11,34 @@ function msgcontains(message, keyword)
...
 
 	local a = 0
 	if inBackpacks then
-		local container, b = doCreateItemEx(backpack, 1), 1
+		local container, b = Game.createItem(backpack, 1), 1
 		for i = 1, amount do
-			local item = doAddContainerItem(container, itemid, subType)
-			if table.contains({(getContainerCapById(backpack) * b), amount}, i) then
-				if doPlayerAddItemEx(cid, container, ignoreCap) ~= RETURNVALUE_NOERROR then
+			local item = doAddContainerItem(container, itemid, subType) -- ???
```
Comment (WibbenZ) [inline data/npc/lib/npc.lua]: Same thing here as above;
"
Why did you add "-- ???"
And why did you not change the compat function? https://github.com/otland/forgottenserver/blob/master/data/lib/compat/compat.lua#L583-L594
"

File: data/npc/lib/npc.lua [hunk 3]
```diff
@@ -56,33 +57,37 @@ function doNpcSellItem(cid, itemid, amount, subType, ignoreCap, inBackpacks, bac
 end
 
 local func = function(cid, text, type, e, pcid)
-	if isPlayer(pcid) then
-		doCreatureSay(cid, text, type, false, pcid, getCreaturePosition(cid))
-		e.done = TRUE
+	local player = Creature(cid)
```
Comment (WibbenZ) [inline data/npc/lib/npc.lua]: Move this to below ```if player2 then```

File: data/npc/lib/npc.lua [hunk 4]
```diff
@@ -56,33 +57,37 @@ function doNpcSellItem(cid, itemid, amount, subType, ignoreCap, inBackpacks, bac
...
+		player:say(text, type, false, pcid, player:getPosition())
+		e.done = true
 	end
 end
 
 function doCreatureSayWithDelay(cid, text, type, delay, e, pcid)
-	if isPlayer(pcid) then
-		e.done = FALSE
+	local player = Player(pcid)
+	if player then
```
Comment (WibbenZ) [inline data/npc/lib/npc.lua]: Since player is only used once you can use ```if Player(pcid) then```

File: data/npc/lib/npcsystem/modules.lua [hunk 5]
```diff
@@ -986,35 +988,36 @@ if Modules == nil then
...
-			doPlayerSendCancel(cid, msg)
+			player:sendCancelMessage(msg)
 			self.npcHandler.talkStart[cid] = os.time()
 			return false
 		end
 	end
 
 	-- Callback for requesting a trade window with the NPC.
-	function ShopModule.requestTrade(cid, message, keywords, parameters, node)
+	function ShopModule.requestTrade(cid, message, keywords, parameters, node)
```
Comment (WibbenZ) [inline data/npc/lib/npcsystem/modules.lua]: Remove trailing tab

**Comment by DSpeichert** [comment]
The source repository seems to be removed and we can't accept that as-is.
I'm closing this and we will replicate some of those changes in other PRs.
